wallet-new/integration/Main.hs

```diff
@@ -55,18 +51,23 @@ main = do
     _ <- runActionCheck
         walletClient
         walletState
-        actionDistr
+        actionDistribution
 
     pure ()
   where
-    actionDistribution :: Either WalletTestError ActionProbabilities
+    actionDistribution :: ActionProbabilities
     actionDistribution = do
-        postWalletProb <- createProbability 50
-        getWalletProb  <- createProbability 50
-
-        pure $ (PostWallet, postWalletProb) :|
-             [ (GetWallet, getWalletProb)
-             ]
+        (PostWallet, Weight 1) :| fmap (\x -> (x, Weight 1)) [minBound .. maxBound]
+
+initialWalletState :: WalletClient IO -> IO  WalletState
+initialWalletState wc = do
+    _wallets <- either throwM (pure . wrData) =<< getWallets wc
```

Comment:
I was getting test failures where we'd do `GetWallets`, and there would *gasp* be some wallets already! so I initialize the state of the wallet with whatever the API returns. I'll likely need to get more of the stuff from the wallet initial state as well.

Reply:
Hm that's strange